### PR TITLE
Set `app_msg` to empty string by default

### LINKED ISSUE #35508: [Serve] Error status message isn't cleared after service redeployed correctly
### What happened + What you expected to happen

Status message isn't cleared after service goes from `DEPLOY_FAILED` to `RUNNING`

### Versions / Dependencies

master (commit 658c8e3)

### Reproduction script

1. Call `serve deploy` with incorrect runtime_env
```
runtime_env:
    working_dir: "https://not_a_valid_website.com/HEAD.zip"
```

Screenshot 2023-05-18 at 11 32 46 AM

2. Call `serve deploy` with correct runtime_env, however error status message remains

```
    working_dir: "https://github.com/ray-project/serve_config_examples/archive/HEAD.zip"
```

Screenshot 2023-05-18 at 11 33 10 AM

### Issue Severity

None

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

The `ApplicationState` currently updates its `_status` directly without necessarily updating the corresponding message. This causes stale messages to sometimes appear with updated statuses. This change sets the `app_msg` to an empty string by default, so the message is cleared or updated whenever the status is changed.

## Related issue number

Closes #35508

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change adds a check to `test_update_app_deploy_failed` to make sure the error message is cleared.
     - I also manually tested that the message is cleared when the `runtime_env` is originally invalid.

### CODE REVIEW

File: python/ray/serve/_private/application_state.py [hunk 1]
```diff
@@ -202,42 +202,60 @@ def update(self):
...
                 except RayTaskError as e:
-                    self._status = ApplicationStatus.DEPLOY_FAILED
                     # NOTE(zcin): we should use str(e) instead of traceback.format_exc()
                     # here because the full details of the error is not displayed
                     # properly with traceback.format_exc(). RayTaskError has its own
                     # custom __str__ function.
-                    self._app_msg = f"Deploying app '{self._name}' failed:\n{str(e)}"
+                    self._update_status(
+                        ApplicationStatus.DEPLOY_FAILED,
+                        app_msg=(f"Deploying app '{self._name}' failed:\n{str(e)}"),
```
Comment (edoakes) [inline python/ray/serve/_private/application_state.py:211]:
```suggestion
                        app_msg=f"Deploying app '{self._name}' failed:\n{str(e)}",
```

**Review by edoakes — COMMENTED** [review]
nit: `app_msg` should be `status_msg` ?

**Comment by edoakes** [comment]
LGTM. Leave approval to @zcin.

Please manually test the workflow I described in the github issue before merging.

**Comment by shrekris-anyscale** [comment]
> nit: app_msg should be status_msg ?

Good catch, I changed it.

> Please manually test the workflow I described in the github issue before merging.

Sure, I manually tested the `runtime_env` workflow, and the message gets cleared correctly with this change.

**Review by zcin — APPROVED** [review]
LGTM!

**Comment by shrekris-anyscale** [comment]
The test failures are unrelated:

* Windows Build & Test: [`test_task_events`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/23066#018849e2-38a0-43dd-a044-7674db378955/3297-3835) is broken on `master`:

Screen Shot 2023-05-23 at 4 42 03 PM

* Doc tests: [`loading data`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/23066#018849f3-e3f8-4609-a77c-9bb55ec4b42b/1429-3223) and [`computer vision`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/23066#018849f3-e3f8-4609-a77c-9bb55ec4b42b/1429-3218) are broken on `master`:

Screen Shot 2023-05-23 at 4 42 57 PM

**Comment by shrekris-anyscale** [comment]
@edoakes @architkulkarni– This change is ready to merge.

**Pull request merged**